golang/cosmos/types/ibc_packet.go

```diff
@@ -0,0 +1,64 @@
+package types
+
+import (
+	"encoding/json"
+
+	clienttypes "github.com/cosmos/ibc-go/v6/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v6/modules/core/04-channel/types"
+	"github.com/cosmos/ibc-go/v6/modules/core/exported"
+)
+
+var _ json.Marshaler = IBCPacket{}
+var _ json.Unmarshaler = (*IBCPacket)(nil)
+var _ exported.PacketI = IBCPacket{}
+
+type IBCPacket struct {
+	channeltypes.Packet
+}
+
+func (p IBCPacket) MarshalJSON() ([]byte, error) {
+	return packageCdc.MarshalJSON(&p.Packet)
+}
+
+func (p *IBCPacket) UnmarshalJSON(bz []byte) error {
+	return packageCdc.UnmarshalJSON(bz, &p.Packet)
+}
+
+func MakeIBCPacket(
+	data []byte,
+	sequence uint64,
+	sourcePort string,
+	sourceChannel string,
+	destPort string,
+	destChannel string,
+	timeoutHeight clienttypes.Height,
+	timeoutTimestamp uint64,
+) IBCPacket {
+	cp := channeltypes.NewPacket(
+		data, sequence,
+		sourcePort, sourceChannel,
+		destPort, destChannel,
+		timeoutHeight, timeoutTimestamp,
+	)
+	return IBCPacket{Packet: cp}
+}
+
+func CopyToChannelPacket(packet exported.PacketI) channeltypes.Packet {
```

Comment:
Is there a reason to use `channeltypes.Packet` anywhere in our code? Are we worried of any IBC code that wouldn't be able to handle our IBCPacket wrapper? Or some IBC code that expects the original JSON marshalling?

Reply:
We're not worried, but there are upstream IBC methods and interfaces that demand `channeltypes.Packet`.  I've tried to change every part of our code that doesn't have this dependency, but I may have missed some.